src/main/java/org/embl/mobie/io/ome/zarr/util/N5ZarrImageReader.java

```diff
@@ -42,12 +44,41 @@ default String getDimensionSeparator(HashMap<String, JsonElement> attributes) {
         return dimSep == null ? DEFAULT_SEPARATOR : dimSep.getAsString();
     }
 
-    //////TODO:
-    default void getDimensions(HashMap<String, JsonElement> attributes) {
+    default void getDimensions(HashMap<String, JsonElement> attributes) throws IllegalArgumentException {
         JsonElement multiscales = attributes.get("multiscales");
-        if (multiscales != null) {
+       if (multiscales == null) {
+           return;
+       }
+       String version = multiscales.getAsJsonArray().get(0).getAsJsonObject().get("version").getAsString();
+        if (version.equals("0.3")) {
             JsonElement axes = multiscales.getAsJsonArray().get(0).getAsJsonObject().get("axes");
             setAxes(axes);
+        } else if (version.equals("0.4")) {
+            JsonArray axes = multiscales.getAsJsonArray().get(0).getAsJsonObject().get("axes").getAsJsonArray();
+            int index = 0;
+            List<ZarrAxe> zarrAxes = new ArrayList<>();
```

Comment:
@KateMoreva `ZarrAxe` is this supposed to be the singular of `ZarrAxes`?
If so, correct would be singular: `ZarrAxis`, plural: `ZarrAxes`.

Reply:
Yes, my mistake. Good point, thank you!